src/darwin/Framework/CHIP/MTRDevice_Concrete.mm

```diff
@@ -3319,6 +3319,133 @@ - (BOOL)_attributeDataValue:(NSDictionary *)one isEqualToDataValue:(NSDictionary
     return [one[MTRTypeKey] isEqual:theOther[MTRTypeKey]] && ((one[MTRValueKey] == theOther[MTRValueKey]) || [one[MTRValueKey] isEqual:theOther[MTRValueKey]]);
 }
 
+// _attributeDataValue:isEquivalentToDataValue: differs from
+// _attributeDataValue:isEqualToDataValue in that it does not insist on the
+// order of fields in structs matching between the two values.  While in theory
```

Comment:
Should this potentially be an asymmetric test where the actual value can have additional struct fields that are not included in the expected value?

Reply:
Hmm.  That is a good question.  As in, allow specifying only the parts of the struct you care about in terms of doing the comparison?

I'll think about that a bit as a followup, but for now I think testing for having the same fields is good.

Reply:
It's also worth thinking a bit about whether list order should be enforced, by the way.  At some point we might need some configuration flags for this or something.  But for now, let's stick to this.